src/features/auth/Login.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect } from "react";
+import { useState } from "react";
+import toast from "react-hot-toast";
+import { useDispatch, useSelector } from "react-redux";
+import { Link, useNavigate } from "react-router-dom";
+import { Input } from "../../components";
+import { loginUserThunk } from "./authSlice";
+
+export const Login = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const [passWordVisible, setPassWordVisible] = useState(false);
+  const { loading, token } = useSelector((state) => state.authReducer);
+
+  // username: "adarshbalika",
+  // password: "adarshBalika123",
+  // username: "ajay",
+  // password: "ajay123",
+  const [loginUser, setLoginUser] = useState({
+    userName: "",
+    password: "",
+  });
+
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    setLoginUser({
+      ...loginUser,
+      [name]: value,
+    });
+  };
+
+  const handleUserLogin = (e) => {
+    e.preventDefault();
+    dispatch(
+      loginUserThunk({
+        username: loginUser.userName,
+        password: loginUser.password,
+      })
+    )
+      .unwrap()
+      .then(({ foundUser }) => {
+        toast.success(`Hey Welcome Back ${foundUser.username}`);
+      })
+      .catch((error) => {
+        toast.error("Some thing went wrong");
+      });
```

Comment:
Appreciation: you are handling success and error state with then(), unwrap(), and catch method.